File: software/examples/vbnet/ExamplePrintVoltages.vb
Imports System
Imports Tinkerforge

Module ExamplePrintVoltages
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XXYYZZ" ' Change XXYYZZ to the UID of your HAT Brick

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim hat As New BrickHAT(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Get current get voltages
        Dim voltageUSB, voltageDC As Integer

        hat.GetVoltages(voltageUSB, voltageDC)

        Console.WriteLine("Voltage USB: " + (voltageUSB/1000.0).ToString() + " V")
        Console.WriteLine("Voltage DC: " + (voltageDC/1000.0).ToString() + " V")

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module

File: software/examples/go/example_print_voltages.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/hat_brick"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
)

const ADDR string = "localhost:4223"
const UID string = "XXYYZZ" // Change XXYYZZ to the UID of your HAT Brick.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	hat, _ := hat_brick.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Get current get voltages.
	voltageUSB, voltageDC, _ := hat.GetVoltages()

	fmt.Printf("Voltage USB: %f V\n", float64(voltageUSB)/1000.0)
	fmt.Printf("Voltage DC: %f V\n", float64(voltageDC)/1000.0)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()
}
